src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -59,16 +59,10 @@
     /**
      * Constructs an instance of the DiskBasedAsyncCache at the specified directory.
      *
-     * @param rootDirectory The root directory of the cache.
+     * @param rootDirectorySupplier The root directory supplier of the cache.
      */
-    public DiskBasedAsyncCache(final File rootDirectory, int maxCacheSizeInBytes) {
-        mRootDirectorySupplier =
-                new FileSupplier() {
-                    @Override
-                    public File get() {
-                        return rootDirectory;
-                    }
-                };
+    private DiskBasedAsyncCache(final FileSupplier rootDirectorySupplier, int maxCacheSizeInBytes) {
```

Comment:
Do we really need to mark the FileSupplier parameter `final`? I feel like it isn't needed, especially as a constructor parameter: it's unlikely the variable will be re-assigned within the constructor.

Reply:
Yeah, not sure why that's there. Don't think it's necessary.